components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -4114,6 +4114,58 @@ public String getServiceProviderNameByClientId(String clientId, String clientTyp
         return applicationName;
     }
 
+    /**
+     * @param inboundKey inbound key
+     * @param inboundType inbound type
+     * @param tenantDomain tenant domain
+     * @return application resource id
+     * @throws IdentityApplicationManagementException
+     */
+    public String getApplicationResourceIDByInboundKey(String inboundKey, String inboundType, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        int tenantID = MultitenantConstants.SUPER_TENANT_ID;
```

Comment:
why do we need to initalize this value here?

Reply:
fixed